src/utility.ts

```diff
@@ -459,17 +459,24 @@ export function addFileProtocol(
  * @param filePath
  */
 export function removeFileProtocol(filePath: string): string {
-  if (process.platform === "win32") {
-    return filePath
-      .replace(/^(file|vscode-resource|vscode-webview-resource)\:\/+/, "")
-      .replace(/^file\/+/, "");
-  } else {
-    return (
-      filePath
-        .replace(/^(file|vscode-resource|vscode-webview-resource)\:\/+/, "")
-        .replace(/^file\/+/, "") + "/"
-    );
-  }
+  // See https://regex101.com/r/YlpEur/8/
+  // - "file://///a///b//c/d"                   ---> "a///b//c/d"
+  // - "vscode-resource://///file///a///b//c/d" ---> "file///a///b//c/d"
+  const regex = /^(?:(?:file|(vscode)-(?:webview-)?resource|vscode--resource):\/+)(.*)/m;
+
+  return filePath.replace(regex, (m, isVSCode, rest) => {
+    if (isVSCode) {
+      // For vscode urls -> Remove `://file/...` host
+      rest = rest.replace(/^file\/+/, "/");
```

Comment:
Hi @gabyx , I feel this line should be `rest = rest.replace(/^file\/+/, "");`?

Reply:
For example, on Windows vscode we will get `vscode-resource://file///C:/a/b/c.txt`, and we want to get `C:/a/b/c.txt` in the end but your code now will give us `/C:/a/b/c.txt`. Thank you.

Reply:
Haha correct :-) Still buggy. Wait I test this function again...

Reply:
OK now IMO its correct.